include/envoy/http/header_map.h

```diff
@@ -22,6 +22,10 @@ class LowerCaseString {
 
   const std::string& get() const { return string_; }
   bool operator==(const LowerCaseString& rhs) const { return string_ == rhs.string_; }
+  LowerCaseString& operator=(const LowerCaseString& rhs) {
```

Comment:
per offline convo make config a pointer to a struct and remove operator= from here (so we don't introduce accidental copies).